qt/python/mantidqt/project/recovery/test/test_projectrecovery.py

```diff
@@ -0,0 +1,387 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantidqt package
+#
+
+import unittest
+import os
+import getpass
+import tempfile
+import time
+import shutil
+import sys
+import json
+
+from mantidqt.project.recovery.projectrecovery import ProjectRecovery, SAVING_TIME_KEY, NO_OF_CHECKPOINTS_KEY, \
+    RECOVERY_ENABLED_KEY
+from mantid.kernel import ConfigService
+from mantid.simpleapi import CreateSampleWorkspace, GroupWorkspaces
+from mantid.api import AnalysisDataService as ADS
+
+if sys.version_info.major >= 3:
+    from unittest import mock
+else:
+    import mock
+
+
+class FakeEncoder(object):
+    tags = ["FakeEncoder"]
+
+    @staticmethod
+    def encode(_, __):
+        return {}
+
+
+class ProjectRecoveryTest(unittest.TestCase):
+    def setUp(self):
+        self.globalfigmanager = mock.MagicMock()
```

Comment:
globalfigmanager is just passed through to the projectsaver object, I mocked it out so if it did change it wouldn't break the test, the tests for that are handled in project saver.

Reply:
I can't see it being passed directly to ProjectSaver, only a retrieval of the figures `plots = self.gfm.figs`. 

It seems that if this `.figs` access changed the test should break, as the original code would no longer call `project_saver.save_project` successfully either.

Reply:
Hmmmm... Ahh yes I changed the implementation of project save at some point.

Reply:
I have an issue with being able to check if .figs works unfortunately since we do not want to have a dependence on the global figure manager class in the workbench package in mantidqt, this will not work.